src/SROCK_utils.jl

```diff
@@ -34,7 +34,7 @@ function maxeig!(integrator, cache::StochasticDiffEqConstantCache)
     z *= quot
   else
     dz_u = pert
-    z = dz_u*ones(z)
+    z = dz_u .* (false .* z .+ one(eltype(z)))
```

Comment:
why is it like this? `z = dz_u*one(z)`?

Reply:
```
julia> z = rand(4,5)
4×5 Array{Float64,2}:
 0.0543206  0.0619733  0.888743  0.822685  0.417369
 0.0877441  0.67813    0.704411  0.493939  0.239282
 0.444318   0.78979    0.3604    0.586677  0.402327
 0.344372   0.710735   0.400021  0.646333  0.178213

julia> ones(z)
ERROR: MethodError: no method matching ones(::Array{Float64,2})
Closest candidates are:
  ones(::Union{Integer, AbstractUnitRange}...) at array.jl:448
  ones(::Type{T}, ::Union{Integer, AbstractUnitRange}...) where T at array.jl:449
  ones(::Tuple{Vararg{Union{Integer, AbstractUnitRange},N} where N}) at array.jl:450
  ...
Stacktrace:
 [1] top-level scope at none:0

```

Reply:
Yes, using `one(z)` would work though.

Reply:
`one(z)` gives the identity matrix. We want one at all places.

Reply:
And also the matrix `z` might not be square.

Reply:
oh oops, yeah that's a v1.0 thing.